base/linalg/dense.jl

```diff
@@ -343,31 +345,16 @@ function ^{T}(A::AbstractMatrix{T}, p::Real)
         retmat = retmat * full(powm(UpperTriangular(float.(A)), real(p - floor(p))))
         d = diag(A)
     else
-        S,Q,d = schur(complex(A))
+        S,Q,d = schur(complex(full(A)))
```

Comment:
Shouldn't the `schur` function have a method that calls `full` if necessary?

Why wouldn't you just do `schur(A)`?